table/tables/tables.go

```diff
@@ -323,6 +323,20 @@ func (t *Table) rebuildIndices(ctx context.Context, h int64, touched []bool, old
 	return nil
 }
 
+func (t *Table) delDirtyIndexKey(createdIdx int, recordID int64, r []interface{}, txn kv.Transaction) {
+	for i, v := range t.indices {
+		if v == nil {
+			continue
+		}
+
+		if i >= createdIdx {
+			break
+		}
+		colVals, _ := v.FetchValues(r)
+		_ = v.X.Delete(txn, colVals, recordID)
```

Comment:
This maybe not correct. You want to delete dirty data, but this may delete the data in kv.
For example there is a row [2, 2] already in kv store and you insert [2, 1] and meet ErrKeyExists. In this case, you may delete the correct index for row [2, 2].
I think it is better to check constraints before creating index.